From 0cb98710913f2c9e2ca2e4c51eab08716295ad1f Mon Sep 17 00:00:00 2001
From: Mark Rushakoff <mark@strange.love>
Date: Fri, 25 Oct 2024 15:35:39 -0400
Subject: [PATCH 1/2] chore: use cosmossdk.io/log/slog

We still need a way to correctly inject this into the config, but at
least now we don't have to maintain a patch for an slog implementation.
---
 server/v2/commands.go | 10 ++++++----
 server/v2/go.mod      |  1 +
 simapp/v2/go.mod      |  1 +
 3 files changed, 8 insertions(+), 4 deletions(-)

diff --git a/server/v2/commands.go b/server/v2/commands.go
index c64fbc1f0d..4e07773667 100644
--- a/server/v2/commands.go
+++ b/server/v2/commands.go
@@ -3,6 +3,7 @@ package serverv2
 import (
 	"context"
 	"errors"
+	"log/slog"
 	"os"
 	"os/signal"
 	"path/filepath"
@@ -15,6 +16,7 @@ import (
 
 	"cosmossdk.io/core/transaction"
 	"cosmossdk.io/log"
+	cslog "cosmossdk.io/log/slog"
 )
 
 // Execute executes the root command of an application.
@@ -205,10 +207,10 @@ func configHandle[T transaction.Tx](s *Server[T], cmd *cobra.Command) error {
 		return err
 	}
 
-	logger, err := NewLogger(v, cmd.OutOrStdout())
-	if err != nil {
-		return err
-	}
+	// Override the logger to slog and stderr so we can stream it to stderr during test.
+	logger := cslog.NewCustomLogger(
+		slog.New(slog.NewTextHandler(os.Stderr, nil)),
+	)
 
 	return SetCmdServerContext(cmd, v, logger)
 }
diff --git a/server/v2/go.mod b/server/v2/go.mod
index 3a17088382..82f738fa42 100644
--- a/server/v2/go.mod
+++ b/server/v2/go.mod
@@ -4,6 +4,7 @@ go 1.23
 
 replace (
 	cosmossdk.io/api => ../../api
+	cosmossdk.io/log => ../../log
 	cosmossdk.io/server/v2/appmanager => ./appmanager
 	cosmossdk.io/server/v2/stf => ./stf
 	cosmossdk.io/store/v2 => ../../store/v2
diff --git a/simapp/v2/go.mod b/simapp/v2/go.mod
index 130734beed..1fc6926c75 100644
--- a/simapp/v2/go.mod
+++ b/simapp/v2/go.mod
@@ -250,6 +250,7 @@ require (
 replace (
 	cosmossdk.io/client/v2 => ../../client/v2
 	cosmossdk.io/collections => ../../collections
+	cosmossdk.io/log => ../../log
 	cosmossdk.io/tools/confix => ../../tools/confix
 	cosmossdk.io/x/accounts => ../../x/accounts
 	cosmossdk.io/x/accounts/defaults/base => ../../x/accounts/defaults/base
-- 
2.47.0

